client/client.cpp

```diff
@@ -1,23 +1,78 @@
+#include <iostream>
+#include <sodium.h>
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/transport/TSocket.h>
+#include <thrift/transport/TTransportUtils.h>
+
+#include "../constants/network.h"
 #include "../crypto/encryption_engine.h"
+#include "../gen-cpp/Operation_types.h"
+#include "../gen-cpp/RPC.h"
 #include "../host/redis.h"
-#include <iostream>
-using namespace std;
+
+using namespace apache::thrift;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+
+Operation genRandOperation() {
+	float r = (float) rand() / RAND_MAX;
+	int key = rand() % KEY_MAX;
+  
+	Operation op;
+    
+	op.__set_op(r < 0.5 ? "PUT" : "GET");
+  op.__set_key(std::string(std::to_string(key)));
+	if(op.op == "GET") {
+		char value[VALUE_SIZE];
+		randombytes_buf(value, VALUE_SIZE);
+		op.__set_value(std::string(value));
+	}
+	return op;
+}
+
+void client(std::vector <float>* latencies) {
+	std::shared_ptr<TTransport> socket(new TSocket(HOST_IP, HOST_PORT));
```

Comment:
See my comment (nit) on initializing shared ptrs here
https://github.com/ySteinhart1/ORTOA/pull/6/files#r1359689724

Reply:
I would prefer the `make_shared` approach since it only uses one memory allocation. I'll have that changed in new commit